samples/msgext-unfurling-ac-loop-components/nodejs/bots/teamsBot.js

```diff
@@ -0,0 +1,99 @@
+const { TeamsActivityHandler, CardFactory } = require("botbuilder");
+
+const card = CardFactory.adaptiveCard({
+  "$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
+  "type": "AdaptiveCard",
+  "version": "1.6",
+  "metadata": {
+      "webUrl": "https://github.com/OfficeDev/Microsoft-Teams-Samples/tree/main"
+  },
+  "body": [
+      {
+      "type": "TextBlock",
+      "text": "Adaptive Card-based Loop component",
+      "size": "Large",
+      "weight": "Bolder"
+      },
+      {
+      "type": "Container",
+      "items": [
+          {
+          "type": "TextBlock",
+          "text": "These samples are designed to help understand Microsoft Teams platform capabilities and scenarios(Bots,Tabs,Message extensions,Meeting extensions,Personal apps,Webhooks and connectors)",
+          "weight": "bolder",
+          "size": "medium"
+          }
+      ]
+   }
+  ]
+});
+
+class TeamsBot extends TeamsActivityHandler {
+  constructor() {
+    super();
+  }
+
+  // Msgext-link-unfurling
+  handleTeamsAppBasedLinkQuery() {
+
+    const attachment = card;
+
+    attachment.preview = {
```

Comment:
Here you are changing the reference of the card object as well. Is it not causing any problem. Actually this would add preview property to the card object as well. Then when you use the card in later parts it would have preview property already defined.

Reply:
Same applies to all the places where you reference the card object.

Reply:
Fixed: handleTeamsMessagingExtensionQuery needs the required "attachment. preview".